### PR TITLE
Use /run/shm for clang transients

### PR DESCRIPTION
**Pull request by waifairer**

### CODE REVIEW

File: tinygrad/runtime/ops_clang.py [hunk 1]
```diff
@@ -14,12 +14,10 @@
 class ClangProgram:
   def __init__(self, name:str, prg:str):
     prg = CLANG_PROGRAM_HEADER + prg
-    # TODO: is there a way to not write this to disk?
-    fn = f"{tempfile.gettempdir()}/clang_{hashlib.md5(prg.encode('utf-8')).hexdigest()}.{args['ext']}"
+    # /run/shm is a in-mem tmpfs, so this is fast
+    fn = f"/run/shm/clang_{hashlib.md5(prg.encode('utf-8')).hexdigest()}.{args['ext']}"
```
Comment (wozeparrot) [inline tinygrad/runtime/ops_clang.py:18]: is `/run/shm` standard? it doesn't exist on my system, but `/dev/shm` does

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/runtime/ops_clang.py              3               5
------------------------------------------------------------
lines added in the tinygrad folder: -2
```

**Comment by geohot** [comment]
This is not the sort of fix I want for this, and it completely ignores the race condition. `/tmp` is also a ramdisk.

**Comment by waifairer** [comment]
Understood. I believe that solving this without a filesystem call involves extending our own version of `dlopen`. But I will spend time understanding the race condition better before providing another potential solution. @geohot

File: tinygrad/runtime/ops_clang.py (hunk 1 above)
Comment (waifairer) [reply tinygrad/runtime/ops_clang.py:18]: Standards changed but I'm not sure which version it happened.

Comment (waifairer) [reply tinygrad/runtime/ops_clang.py:18]: https://askubuntu.com/questions/169495/what-are-run-lock-and-run-shm-used-for

**Pull request closed without merging**